src/Storages/StorageS3.cpp

```diff
@@ -1165,12 +1156,12 @@ static std::shared_ptr<StorageS3Source::IIterator> createFileIterator(
     {
         /// Iterate through disclosed globs and make a source for each file
         return std::make_shared<StorageS3Source::DisclosedGlobIterator>(
-            *configuration.client, configuration.url, query, virtual_columns,
+            *configuration.client, configuration.url, predicate, virtual_columns,
             local_context, read_keys, configuration.request_settings, file_progress_callback);
     }
     else
     {
-        Strings keys = filterKeysForPartitionPruning(configuration.keys, configuration.url.bucket, virtual_columns, filter_dags, local_context);
+        Strings keys = filterKeysForPartitionPruning(configuration.keys, configuration.url.bucket, virtual_columns, predicate, local_context);
```

Comment:
I don't think we need functions `filterKeysForPartitionPruning` and `getBlockWithVirtuals` added in https://github.com/ClickHouse/ClickHouse/pull/56668. We can use `VirtualColumnUtils::filterByPathOrFile` in `StorageS3Source::KeysIterator` as it was before